### PR TITLE
Fix fractional unroll_length

### PR DESCRIPTION
**Pull request by emailweixu**
There are two bugs for fractional_unroll:
1. replay_buffer is None for train_from_replay_buffer
2. missing summary for unroll under certain condition

Also support interactionless training at the end. The main purpose of this is to train more on the last environmental steps.

### CODE REVIEW

**Review by le-horizon — CHANGES_REQUESTED** [review]
Almost good.

Need to update `common.compute_summary_or_eval_interval` which still relies on the two termination conditions being exclusive.

Thanks,
Le

**Comment by emailweixu** [comment]
> Almost good.
> 
> Need to update `common.compute_summary_or_eval_interval` which still relies on the two termination conditions being exclusive.
> 
> Thanks, Le

Good catch. Fixed.

**Review by le-horizon — DISMISSED** [review]
LG thanks for the fixes, Wei.

File: alf/trainers/policy_trainer.py [hunk 1]
```diff
@@ -606,7 +613,7 @@ def _train(self):
             self._trainer_progress.update(iter_num, total_time_steps)
 
             if ((self._num_iterations and iter_num >= self._num_iterations)
-                    or (self._num_env_steps
+                    or (not self._num_iterations and self._num_env_steps
```
Comment (hnyu) [inline alf/trainers/policy_trainer.py]: no need to `and self._num_env_steps` because it has to be >0 in this case.

**Review by hnyu — APPROVED** [review]

**Pull request merged**